colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -447,4 +456,7 @@ class ServiceClient[P <: Protocol](
   def disconnect() {
     connection.disconnect()
   }
+
+  override def update(addresses: Seq[InetSocketAddress]): Unit =
+    throw new NotImplementedError("Can not directly update service client")
```

Comment:
`ServiceClient` cannot be updated directly. Updates are handled by the load balancing layer.